nifi-nar-bundles/nifi-jwt-bundle/nifi-jwt-processors/src/main/java/org/apache/nifi/processors/jwt/UnsignJWT.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.bouncycastle.util.encoders.Base64;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.List;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map.Entry;
+
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@SideEffectFree
+@Tags({ "jwt", "json", "flowfile", "unsign" })
+@CapabilityDescription("Validates and unsigns a JWT either from the flowfile body or from an attribute. The JWT is unsigned "
+                + "using a directory of public RSA keys specified by the \"Approved Public Keys\" property. By default the "
+                + "processor will read the JWT from the content of the flowfile and rewrite the contents with the JWT claims "
+                + "in JSON format. If an attribute name is given as the \"JWT Attribute Name\" property then the content of "
+                + "that attribute will be unsigned and the JWT claims will be written as \"jwt.XXX\" attributes.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "jwt.header", description = "If the flow file is routed to the success relationship "
+                + "the attribute will contain the the header of the JWT."),
+        @WritesAttribute(attribute = "jwt.footer", description = "If the flow file is routed to the success relationship "
+                + "the attribute will contain the footer of the JWT."),
+            @WritesAttribute(attribute = "jwt.XXX", description = "If the flow file is routed to the success relationship "
+                + "and the JWT was unsugned from an attribute, these attributes will contain the claims of the JWT."),
+        @WritesAttribute(attribute = "failure.reason", description = "If the flow file is routed to the failure relationship "
+                + "the attribute will contain the error message resulting from the validation failure.") })
+@SeeAlso({ SignJWT.class })
+public class UnsignJWT extends AbstractProcessor {
+
+    public static final PropertyDescriptor PUBLIC_KEYS_PATH = new PropertyDescriptor.Builder().name("PUBLIC_KEYS_PATH")
```

Comment:
Again, this could be expanded to allow direct population of the public key, and many public keys are not provided in `.pub` files -- perhaps the default extension filter could be `*.pub` but allow for a regex like `*.pub|*.pem`, etc.

Reply:
What would be the best way to go about setting multiple public keys in the property? Just as multiple separate properties?